addons/account/models/account_analytic_distribution_model.py

```diff
@@ -27,9 +27,14 @@ def _create_domain(self, fname, value):
         if not fname == 'account_prefix':
             return super()._create_domain(fname, value)
 
+    def _get_fields_to_check(self):
+        fields_list = super()._get_fields_to_check()
+        fields_list.extend(['account_prefix', 'product_id', 'product_categ_id'])
+        return fields_list
+
     def _check_score(self, key, value):
-        if key == 'account_prefix':
-            if value.startswith(self.account_prefix):
+        if self[key] and key == 'account_prefix':
```

Comment:
We could add a `if key.endswith('_prefix')` in the `analytic` module, like showed with the `if isinstance(value, (list, tuple))` to avoid this inheritance